core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1868,25 +1802,18 @@ class KafkaApis(val requestChannel: RequestChannel,
                 if (resource.name.isEmpty)
                   throw new InvalidRequestException("Invalid empty resource name")
                 auth.addAcls(immutable.Set(acl), resource)
+
                 if (logger.isDebugEnabled)
                   logger.debug(s"Added acl $acl to $resource")
+
+                new AclCreationResponse(null)
               } catch {
-                case throwable : Throwable => if (logger.isDebugEnabled) {
-                    logger.debug(s"Failed to add acl $acl to $resource", throwable)
-                  }
-                  errors.put(i, throwable)
+                case throwable: Throwable => new AclCreationResponse(throwable)
```

Comment:
Hmm.  Why are we removing the log about failures, but keeping the log about successful additions?  It seems like we should remove both or neither  (maybe you guys have some more insight into whether there are other log messages that will be present in these cases).